src/number_theory/bernoulli_polynomials.lean

```diff
@@ -0,0 +1,163 @@
+  { rintros x hx, split_ifs,
+    { exfalso, apply mem_range_sub_ne_zero hx h, }, refl, },
+  conv_lhs { congr, skip, apply_congr, skip, rw f x H, },
+  cases n, { simp only [one_div, alg_hom.map_smul, power_series.coeff_zero_eq_constant_coeff,
+    add_zero, polynomial.aeval_one, if_congr, mul_one, nat.nat_zero_eq_zero,
+    bernoulli_poly.bernoulli_poly_zero, nat.factorial_zero, if_true, nat.zero_sub,
+    power_series.constant_coeff_X, nat.sub_self, sum_empty, eq_self_iff_true, nat.factorial_one,
+    one_smul, sub_zero, range_zero, power_series.coeff_rescale, ring_hom.map_one, nat.cast_one,
+    div_one, sum_singleton, range_one, mul_zero, pow_zero, power_series.coeff_exp, sum_congr,
+    sub_self, algebra.smul_mul_assoc], },
+  symmetry, rw [sum_eq_single 1],
+  { rw [one_div, alg_hom.map_smul, nat.succ_sub_succ_eq_sub, nat.sub_self, nat.factorial_zero,
+      power_series.coeff_one_X, one_mul, nat.sub_zero, coeff_rescale, coeff_exp, nat.factorial_one,
+      nat.cast_one, div_one],
+    simp only [one_div, alg_hom.map_smul, if_true, eq_self_iff_true, sub_zero, zero_add,
+      ring_hom.map_one, mul_zero, sub_self, algebra.smul_mul_assoc],
+    rw [mul_comm, smul_zero, zero_add],
+    suffices g : (algebra_map ℚ A) (↑n!)⁻¹ * (n.succ : ℚ) • t ^ n =
+      (n.succ : ℚ) • ∑ (x : ℕ) in range n.succ, (↑x!)⁻¹ • ((polynomial.aeval t) (bernoulli_poly x)
+        * (algebra_map ℚ A) (↑(n.succ - x)!)⁻¹),
+    { rw [algebra.mul_smul_comm, algebra.smul_def, algebra.smul_def, mul_eq_mul_left_iff] at g,
+      cases g with g1 g2, { assumption, },
+      { exfalso, apply succ_ne_zero n,
+        simp only [ring_hom.map_nat_cast, cast_succ, ring_hom.map_add, ring_hom.map_one] at g2,
+        norm_cast at *, }, },
+    { have g : (n.succ : ℚ) • t^n = polynomial.aeval t (polynomial.monomial n (n + 1 : ℚ)),
+      { rw [polynomial.aeval_monomial, algebra.smul_def], norm_num, },
+      rw [g, ←sum_bernoulli_poly, smul_sum, alg_hom.map_sum, mul_sum], apply sum_congr, { refl, },
+      { rintros x hx, rw alg_hom.map_smul, rw ←algebra.smul_def,
+        rw [mul_comm ((polynomial.aeval t) (bernoulli_poly x)) _, ←algebra.smul_def, smul_smul,
+          smul_smul, ←smul_assoc, algebra.smul_def, algebra.smul_def, mul_eq_mul_right_iff], left,
+        rw smul_eq_mul, apply congr_arg _,
+        rw [choose_eq_factorial_div_factorial, cast_dvd_char_zero, factorial_succ n, cast_mul,
+          cast_mul, div_eq_inv_mul, mul_inv', ←mul_assoc, mul_comm (↑n!)⁻¹ _,
+          mul_assoc ↑(n.succ) _ _], symmetry,
+        rw [mul_comm ↑(n.succ) _, mul_assoc ((↑x!)⁻¹ * (↑(n + 1 - x)!)⁻¹) _ _, mul_comm ↑(n.succ) _,
+         ←mul_assoc (↑n!)⁻¹ _ _, inv_mul_cancel, one_mul],
+        { norm_num, apply factorial_ne_zero n, },
+        { apply factorial_mul_factorial_dvd_factorial (mem_range_le hx), },
+        { apply mem_range_le hx, }, }, }, },
+  { rintros b hb h, rw coeff_X b, split_ifs,
+    { exfalso, apply h h_1, },
+    { norm_num, }, },
```

Comment:
```suggestion
    { apply zero_mul }, },
```

`norm_num` is for normalising numerals (numbers like 37 or 22/7). But it also tries `simp`, which closes your goal because `zero_mul` is a `simp` lemma. I think it's much quicker to just apply the lemma `simp` finds.